meta-resin-common/recipes-containers/hostapp-update/files/hostapp-update

```diff
@@ -79,16 +81,28 @@ cur_counter=0
 if [ -f "/mnt/sysroot/active/counter" ]; then
 	cur_counter=$(cat /mnt/sysroot/active/counter)
 fi
-echo $(($cur_counter + 1)) > "$SYSROOT/counter.new"
+echo $((cur_counter + 1)) > "$SYSROOT/counter.new"
 sync -f "$SYSROOT"
 mv "$SYSROOT/counter.new" "$SYSROOT/counter"
 sync -f "$SYSROOT"
 
-# Run any defined hooks
-for hook in "/etc/hostapp-update-hooks.d/"*; do
-	[ -e "$hook" ] || break
-	"$hook" "$CONTAINER_ID"
-done
+if [ "$hooks" = 1 ]; then
+	# Run all the defined hooks in the host OS we update to
+	if balena run -v /mnt:/mnt -v /mnt/sysroot/inactive:/mnt/sysroot/active --rm "$HOSTAPP_IMAGE" bash -ec "for hook in /etc/hostapp-update-hooks.d/*; do \$hook; done"; then
```

Comment:
Thinking about this a bit more, I think we shouldn't swap the bind mounts. We should instead set an env var that lets the new version know that it is being run during an update and that's it's not booted right now.

Also, I don't think we should be looping over the `/etc/hostapp-update-hooks.d` directory here. We may change how we do hooks in the future. Let's make the interface a bit more general by saying that the new image must have an executable named `hostapp-update-hooks` that will run whatever needs to be run.

So the command will become something like:

```
 balena run -v /mnt:/mnt --rm -e DURING_UPDATE=1 "$HOSTAPP_IMAGE" hostapp-update-hooks
```

Finally, ours hooks should avoid messing with the sysroots. Our currently defined hooks don't touch it, but I'm mentioning it here so that you have it in mind for the future.

Reply:
Done. Will push with next iteration.

Reply:
Is Petros' comment above tracked as an item to be implemented in the future?

Reply:
No. It's part of this PR.